src/symbolic_memory.ml

```diff
@@ -38,27 +38,30 @@ module M = struct
     | _ -> Log.err {|Unsupported symbolic value reasoning over "%a"|} Expr.pp v
 
   let grow m delta =
-    let delta = Int32.to_int @@ i32 delta in
-    let old_size = m.size * page_size in
-    m.size <- max m.size ((old_size + delta) / page_size)
+    let old_size = Value.I32.mul m.size page_size in
+    let new_size = Value.I32.(div (add old_size delta) page_size) in
+    m.size <-
+      Triop (Ite, Value.I32.gt new_size old_size, new_size, old_size) @: Ty_bool
 
-  let size { size; _ } = Value.const_i32 @@ Int32.of_int (size * page_size)
+  let size { size; _ } = Value.I32.mul size page_size
 
-  let size_in_pages { size; _ } = Value.const_i32 @@ Int32.of_int @@ size
+  let size_in_pages { size; _ } = size
 
   let fill _ = assert false
 
   let blit _ = assert false
 
   let blit_string m str ~src ~dst ~len =
-    (* Always concrete? *)
+    (* This function is only used in memory init so everything will be concrete *)
```

Comment:
OK thus maybe we should change the expected type in the interface to avoid re-using it one day in a symbolic case ?

Reply:
Yes, this would be better to avoid accidents :)

Reply:
Yes, please do. :)

Reply:
OK actually, this is likely to be true in C compiled programs but in a Wasm program one can write a `memory.init` in the middle of a program with a symbolic value. Let's keep it like this.